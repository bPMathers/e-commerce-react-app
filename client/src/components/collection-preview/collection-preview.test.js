import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPreview } from './collection-preview.component';

let wrapper, mockHistory, mockItems, mockItems2, mockMatch;

beforeEach(() => {
  mockItems = [{
    id: 1,
    imageUrl: 'www.testImage1.com',
    price: 10,
    name: "hat1",
  },
  {
    id: 2,
    imageUrl: 'www.testImage2.com',
    price: 15,
    name: "hat2",
  },
  {
    id: 3,
    imageUrl: 'www.testImage3.com',
    price: 20,
    name: "hat3",
  }];

  mockItems2 = [{
    id: 1,
    imageUrl: 'www.testImage1.com',
    price: 10,
    name: "hat1",
  },
  {
    id: 2,
    imageUrl: 'www.testImage2.com',
    price: 15,
    name: "hat2",
  },
  {
    id: 3,
    imageUrl: 'www.testImage3.com',
    price: 20,
    name: "hat3",
  },
  {
    id: 4,
    imageUrl: 'www.testImage4.com',
    price: 20,
    name: "hat4",
  },
  {
    id: 5,
    imageUrl: 'www.testImage5.com',
    price: 20,
    name: "hat5",
  }];

  mockHistory = {
    push: jest.fn()
  }

  mockMatch = {
    path: 'mockPath'
  }

  const mockProps = {
    title: 'hats',
    items: mockItems,
    history: mockHistory,
    match: mockMatch,
    routeName: 'mockRouteName'
  }

  wrapper = shallow(<CollectionPreview {...mockProps} />)
})

describe('CollectionPreview Component', () => {
  it('should render CollectionPreview component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call history.push() with correct argument when TitleContainer is clicked', () => {
    expect.assertions(2);
    wrapper.find('TitleContainer').simulate('click');

    expect(mockHistory.push).toHaveBeenCalled();
    expect(mockHistory.push).toHaveBeenCalledWith('mockPath/mockRouteName');
  });
  it('should render the same number of CollectionItems that it is passed as prop', () => {
    expect(wrapper.find('PreviewContainer').children().length).toBe(mockItems.length);
  });
  it('should not render more than 4 CollectionItems even if passed items prop is of greater length', () => {
    const mockProps2 = {
      title: 'hats',
      items: mockItems2,
      history: mockHistory,
      match: mockMatch,
      routeName: 'mockRouteName'
    }

    const wrapper2 = shallow(<CollectionPreview {...mockProps2} />)
    expect(wrapper2.find('PreviewContainer').children().length).toEqual(4);
  });
})