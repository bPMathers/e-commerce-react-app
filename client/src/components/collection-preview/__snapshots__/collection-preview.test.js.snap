// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview Component should render CollectionPreview component correctly 1`] = `
<styled.div>
  <TitleContainer
    onClick={[Function]}
  >
    HATS
  </TitleContainer>
  <PreviewContainer>
    <Connect(CollectionItem)
      item={
        Object {
          "id": 1,
          "imageUrl": "www.testImage1.com",
          "name": "hat1",
          "price": 10,
        }
      }
      key="1"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 2,
          "imageUrl": "www.testImage2.com",
          "name": "hat2",
          "price": 15,
        }
      }
      key="2"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 3,
          "imageUrl": "www.testImage3.com",
          "name": "hat3",
          "price": 20,
        }
      }
      key="3"
    />
  </PreviewContainer>
</styled.div>
`;
